Added in dashboard.config transform support for nuget & Umbraco packing

jamiepollock opened the pull request:
Hey @leekelleher,
As mentioned over [on Twitter](https://twitter.com/jamiepollock/status/747369264164790272) I have added in transformations for config\dashboard.config to add in the dashboard on package install.

I've tested this with nuget & Umbraco local package install on Umbraco v7.5.0 beta. Seems to work well.

Thanks,
Jamie

Commit:
Added in dashboard.config transform support for nuget & Umbraco packing

leekelleher commented:
Thanks @jamiepollock!  I'll take a look later on today.

leekelleher commented on build/package.proj:
```diff
@@ -132,6 +133,12 @@
 		<Copy SourceFiles="@(NuSpecFile)" DestinationFolder="$(BuildNuGetDir)" />
+
+		<!-- Umbraco -->
+    	<Copy SourceFiles="@(DashboardConfigXDTInstallFile)" DestinationFolder="$(BuildUmbDir)\App_Plugins\BulkUserAdmin\Install" />
```
@jamiepollock Do we need the `*.xdt` in the Umbraco package file?
I thought it would only be for the NuGet package?

jamiepollock commented:
@leekelleher huh, turns it's not needed! Tried it locally and it looks alright. I didn't realise it got added to the package.xml automagically. People are really clever!

mattbrailsford commented:
yea, it's already in the umbraco package manifest here https://github.com/leekelleher/umbraco-bulk-user-admin/blob/develop/build/package.xml#L31

It's an inbuilt umbraco package action, so we just define it in our xml and it just works\* :)

*\* Disclaimer, may not always work

jamiepollock commented:
I'm sorry I missed that when I did my initial PR. :/

mattbrailsford commented:
SO YOU SHOULD BE!

Lol, only joking. I only mentioned it so you knew why :)

Commit:
Removed Umbraco package step as this in included in the package.xml.

jamiepollock commented:
Right, that's removed :)

mattbrailsford commented:
I'm assuming this should be ok, but have you tried installing it in a website project? as well as a web app project?

jamiepollock commented:
@mattbrailsford I'm afraid I haven't. I just found out [how to create a website in VS2015](https://www.youtube.com/watch?v=f9mnzvIpMck).

I'm assuming I'd only test NuGet installation but I can test Umbraco install too.

mattbrailsford commented:
yea, you'd only need to test the nuget file.

In all honesty, it's probably not a big deal if it doesn't work, as a lot don't already, but thought I'd play devils advocate :imp:

jamiepollock commented:
Had a play with this and got stuck trying to build Umbraco (v7.4.3) as a website. :cold_sweat:

This might be a non-starter. The nuget package installed to the website, but couldn't get much further than that.

mattbrailsford commented:
ok, no worries. Probably not worth spending a load of time working on it so long as it works with web apps.

leekelleher commented:
We may need to include an `install.ps1` script too?
(It's something we had to do on Nested Content a while back - ref: https://github.com/leekelleher/umbraco-nested-content/pull/28)

@jamiepollock Don't worry too much about this... unless you want to?

Commit:
Added PowerShell install which should cover our backs!

Mentioned as a potential added feature by @leekelleher here https://github.com/leekelleher/umbraco-bulk-user-admin/pull/1#issuecomment-228765554

jamiepollock commented:
Bingo-bango, that's added in too. Based it on Jeavon's work.

Tested with the following steps:
1. Installing the package via Nuget Package Manager Console: `Install-Package Our.Umbraco.BulkUserAdmin -pre`
2. Checked Install.ps1 was called in trace
3. Excluding /App_Plugins/BulkUserAdmin/ from solution.
4. Called `Update-Package Our.Umbraco.BulkUserAdmin -Reinstall`
5. Checked trace so that "Updating Bulk User Admin App_Plugin files using PS as they have been excluded from the project" appeared
6. ????
7. PROFIT!!

leekelleher commented:
Awesomeness, thanks @jamiepollock.

I'll merge it in now! :sunglasses:

mattbrailsford commented:
:star:

jamiepollock commented:
cheers gents :)
